wpi-many.sh

```diff
@@ -0,0 +1,217 @@
+    exit 4
+fi
+
+if [ "x${GITHUB_USER}" = "x" ]; then
+    GITHUB_USER="${USER}"
+fi
+
+### Script
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# clone or update DLJC
+if [ ! -d "${SCRIPTDIR}/../do-like-javac" ]; then
+    git -C "${SCRIPTDIR}/.." clone https://github.com/kelloggm/do-like-javac
+else
+    git -C "${SCRIPTDIR}/../do-like-javac" pull
+fi
+
+export DLJC="${SCRIPTDIR}/../do-like-javac/dljc"
+    
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
+
+while IFS='' read -r line || [ "$line" ]
+do    
+    REPOHASH=${line}
+
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
+
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
+    REPO_NAME_HASH="${REPO_NAME}-${HASH}"
+
+    # Use repo name and hash, but not owner.  We want
+    # repos that are different but have the same name to be treated
+    # as different repos, but forks with the same content to be skipped.
+    # TODO: consider just using hash, to skip hard forks?
+    mkdir -p "${REPO_NAME_HASH}"
+
+    pushd "${REPO_NAME_HASH}" || exit 5
+    
+    if [ ! -d "${REPO_NAME}" ]; then
+        # this environment variable prevents git from prompting for
+	# username/password if the repository no longer exists
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
+        # skip the rest of the script if cloning isn't successful
+        if [ -d "${REPO_NAME}" ]; then
+           continue
+        fi
+    else
+        rm -rf "${REPO_NAME}/dljc-out"
+    fi
+
+    pushd "${REPO_NAME}" || exit 5
+
+    git checkout "${HASH}"
+
+    OWNER=$(echo "${REPO}" | cut -d / -f 4)
+
+    if [ "${OWNER}" = "${GITHUB_USER}" ]; then
+        ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
+        git remote add unannotated "${ORIGIN}"
+    fi
+
+    REPO_FULLPATH=$(pwd)
+    
+    popd || exit 5
+
+    RESULT_LOG="${OUTDIR}-results/${REPO_NAME_HASH}-wpi.log"
+    touch "${RESULT_LOG}"
+
+    "${SCRIPTDIR}/wpi.sh" -d "${REPO_FULLPATH}" -u "${GITHUB_USER}" -t "${TOUT}" "$@" &> "${RESULT_LOG}"
+
+    popd || exit 5
+
+    # if the result is unusable, we don't need it for data analysis and we can
+    # delete it right away
+    if [ -f "${REPO_FULLPATH}/.unusable" ]; then
+        rm -rf "${REPO_NAME_HASH}" &
+    else
+        cat "${REPO_FULLPATH}/dljc-out/wpi.log" >> "${RESULT_LOG}"
+    fi
+
+    cd "${OUTDIR}" || exit 5
+    
+done <"${INLIST}"
+
+popd || exit 5
+
+unaccounted_for=$(grep -Zvl "no build file found for" "${OUTDIR}-results/*.log" \
+    | xargs -0 grep -Zvl "dljc could not run the Checker Framework" \
+    | xargs -0 grep -Zvl "dljc could not run the build successfully" \
+    | xargs -0 grep -Zvl "dljc timed out for" \
+    | xargs -0 echo)
+
+echo "${unaccounted_for}" > "${OUTDIR}-results/unaccounted_for.txt"
+
+javafiles=$(grep -oh "\S*\.java " "${unaccounted_for}")
```

Comment:
Which Java files are these?  I think not all of them.  Can you clarify?

Reply:
These are all the `.java` files in the projects for which the Checker Framework produced output - that is, the ones that need to be manually inspected. Only those count as part of the experiment's LoC, because those are the only ones for which the CF actually ran.